server/sonar-web/src/main/js/apps/tutorials/onboarding/NewOrganizationForm.js

```diff
@@ -0,0 +1,154 @@
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      done: props.organization != null,
+      loading: false,
+      organization: props.organization || '',
+      unique: true
+    };
+    this.validateOrganization = debounce(this.validateOrganization, 500);
+  }
+
+  componentDidMount() {
+    this.mounted = true;
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  stopLoading = () => {
+    if (this.mounted) {
+      this.setState({ loading: false });
+    }
+  };
+
+  validateOrganization = (organization: string) => {
+    getOrganization(organization).then(response => {
+      if (this.mounted) {
+        this.setState({ unique: response == null });
+      }
+    });
+  };
+
+  sanitizeOrganization = (organization: string) =>
+    organization.replace(/[^a-z0-9-]/, '').replace(/^-/, '');
+
+  handleOrganizationChange = (event: { target: HTMLInputElement }) => {
+    const organization = this.sanitizeOrganization(event.target.value);
+    this.setState({ organization });
+    this.validateOrganization(organization);
+  };
+
+  handleOrganizationCreate = (event: Event) => {
+    event.preventDefault();
+    const { organization } = this.state;
+    if (organization) {
+      this.setState({ loading: true });
+      createOrganization({ key: organization, name: organization }).then(() => {
+        if (this.mounted) {
+          this.setState({ done: true, loading: false });
+          this.props.onDone(organization);
+        }
+      }, this.stopLoading);
+    }
+  };
+
+  handleOrganizationDelete = (event: Event) => {
+    event.preventDefault();
+    const { organization } = this.state;
+    if (organization) {
+      this.setState({ loading: true });
+      deleteOrganization(organization).then(() => {
+        if (this.mounted) {
+          this.setState({ done: false, loading: false, organization: '' });
+          this.props.onDelete();
+        }
+      }, this.stopLoading);
+    }
+  };
+
+  render() {
+    const { done, loading, organization, unique } = this.state;
+
+    return done
+      ? <form onSubmit={this.handleOrganizationDelete}>
+          <span className="spacer-right text-middle">{organization}</span>
+          {loading
+            ? <i className="spinner" />
+            : <button className="button-clean">
+                <i className="icon-delete" />
+              </button>}
+        </form>
+      : <form onSubmit={this.handleOrganizationCreate}>
+          <input
+            autoFocus={true}
+            className="input-super-large spacer-right text-middle"
+            onChange={this.handleOrganizationChange}
+            minLength={2}
+            placeholder={translate('onboarding.organization.placeholder')}
+            required={true}
+            type="text"
+            value={organization}
+          />
+          {loading
+            ? <i className="spinner" />
+            : <button className="text-middle" disabled={!unique}>{translate('create')}</button>}
```

Comment:
Maybe the button should also be disabled when the current organization has less than 2 characters